ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/AssertClientIntegrationTest.java

```diff
@@ -0,0 +1,217 @@
+import org.apache.kafka.streams.StreamsConfig;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+
+@Category({IntegrationTest.class})
+public class AssertClientIntegrationTest {
+
+  private static final PhysicalSchema SCHEMA = PhysicalSchema.from(
+      LogicalSchema.builder()
+          .keyColumn(ColumnName.of("K"), SqlTypes.INTEGER)
+          .valueColumn(ColumnName.of("LONG"), SqlTypes.BIGINT)
+          .build(),
+      SerdeFeatures.of(),
+      SerdeFeatures.of()
+  );
+
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+
+  private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .withProperty(KSQL_STREAMS_PREFIX + StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1)
+      .withProperty(KSQL_DEFAULT_KEY_FORMAT_CONFIG, "JSON")
+      .withProperty(KSQL_HEADERS_COLUMNS_ENABLED, true)
+      .withStaticServiceContext(TEST_HARNESS::getServiceContext)
+      .withProperty(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY, "http://foo:8080")
+      .build();
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain
+      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS))
+      .around(TEST_HARNESS)
+      .around(REST_APP);
+
+
+  @BeforeClass
+  public static void setUpClass() {
+    TEST_HARNESS.ensureTopics("abc");
+    TEST_HARNESS.ensureSchema("abc", SCHEMA, false);
+  }
+
+  private Vertx vertx;
+  private Client client;
+
+  @Before
+  public void setUp() {
+    vertx = Vertx.vertx();
+    client = createClient();
+  }
+
+  @After
+  public void tearDown() {
+    if (client != null) {
+      client.close();
+    }
+    if (vertx != null) {
+      vertx.close();
+    }
+    REST_APP.getServiceContext().close();
+  }
+
+  @Test
+  public void shouldThrowOnAssertNonexistentSchema() {
+    // When
+    final Exception e = assertThrows(
+        ExecutionException.class, // thrown from .get() when the future completes exceptionally
+        () -> client.assertSchema("NONEXISTENT", 3, true, Duration.ofSeconds(3)).get()
+    );
+
+    // Then:
+    assertThat(e.getCause(), instanceOf(KsqlClientException.class));
+    assertThat(e.getCause().getMessage(), containsString("Received 417 response from server"));
+    assertThat(e.getCause().getMessage(), containsString("Schema with subject name NONEXISTENT id 3 does not exist"));
+    assertThat(e.getCause().getMessage(), containsString("Error code: 41700"));
+  }
+
+  @Test
+  public void shouldThrowOnAssertNotExistsNonExistingSchema() {
+    // When
+    final Exception e = assertThrows(
+        ExecutionException.class, // thrown from .get() when the future completes exceptionally
+        () -> client.assertSchema("abc-value", 1, false, Duration.ofSeconds(3)).get()
```

Comment:
As a comment which has nothing to do with this PR.... 

It took me a second to realize that `TEST_HARNESS.ensureSchema("abc", SCHEMA, false);` registers a schema named `abc-value`.

Reply:
added a comment